framework/src/bundle/BundleArchive.h

```diff
@@ -156,11 +156,29 @@ struct BundleArchive : std::enable_shared_from_this<BundleArchive>
 
   std::shared_ptr<BundleResourceContainer> GetResourceContainer() const;
 
+  /**
+   * Increment the number of open resources by one.
+   */
+  void IncrementNumOpenResources() const;
+
+  /**
+   * Decrement the number of open resources by one.
+   */
+  void DecrementNumOpenResources() const;
+
+  /**
+   * Get the number of opened resources.
+   * 
+   * @return the number of opened resources.
+   */
+  unsigned int GetNumOpenResources() const;
+
 private:
   BundleStorage* const storage;
   const std::unique_ptr<Data> data;
   const std::shared_ptr<BundleResourceContainer> resourceContainer;
   const std::string resourcePrefix;
+  mutable unsigned int numOpenResources;
```

Comment:
this should be an ```atomic<uint32_t>``` since a race can occur if more than one thread tries to create ```BundleResource``` object.